variant_remapping_tools/tests/test_reads_to_remapped_variants.py

```diff
@@ -164,9 +167,73 @@ def test_calculate_new_variant_definition(self):
         left_read = self.mk_read(query_name='chr1|48|CAA|C', reference_name='chr2', reference_start=1, reference_end=47, is_reverse=True)
         right_read = self.mk_read(query_name='chr1|48|CAA|C', reference_name='chr2', reference_start=50, reference_end=110, is_reverse=True)
         vcf_rec = ['chr1', '48', '.', 'CAA', 'C']
-        with patch('variant_remapping_tools.reads_to_remapped_variants.fetch_bases', return_value='TTG'):
-            assert calculate_new_variant_definition(left_read, right_read, fasta, vcf_rec) == \
-                   (48, 'TTG', ['G'], {'st': '-'}, None)
+        with patch('variant_remapping_tools.reads_to_remapped_variants.fetch_bases', side_effect=['TTG', 'ATT']):
+            assert next(calculate_new_variants_definition(left_read, right_read, fasta, vcf_rec)) == \
+                   (47, 'ATT', ['A'], {'st': '-'}, None)
+
+    def test_calculate_new_variant_definition2(self):
+        fasta = 'fasta_path'
+        #  reverse strand alignment for Deletion with novel allele
+        # variant:  CTGTG -> C
+        # OLD REF  UUUUUUUUUUUUUUUUUCTGTGDDDDDDDDDDDDDDD
+        # OLD ALT  UUUUUUUUUUUUUUUUUC----DDDDDDDDDDDDDDD
+
+        #                read 2              read 1
+        #          <----------------CACAG<--------------
+        # NEW REF  DDDDDDDDDDDDDDDDACATAGUUUUUUUUUUUUUUU
+        # NEW ALT  DDDDDDDDDDDDDDDDA----GUUUUUUUUUUUUUUU
+        # variant:  ACATA -> A
+        left_read = self.mk_read(query_name='chr1|48|CTGTG|C', reference_name='chr2', reference_start=1, reference_end=46, is_reverse=True)
+        right_read = self.mk_read(query_name='chr1|48|CTGTG|C', reference_name='chr2', reference_start=50, reference_end=110, is_reverse=True)
+        vcf_rec = ['chr1', '48', '.', 'CTGTG', 'C']
+        with patch('variant_remapping_tools.reads_to_remapped_variants.fetch_bases', side_effect=['CATAG', 'ACATA']):
+            var_generator = calculate_new_variants_definition(left_read, right_read, fasta, vcf_rec)
+
+            assert next(var_generator) == \
+                   (46, 'ACATA', ['A'], {'st': '-', 'rac': 'chr2|46|ACACA-ACATA'}, None)
+            assert next(var_generator) == \
+                   (46, 'ACATA', ['ACACA'], {'st': '-', 'rac': 'chr2|46|ACACA-ACATA', 'nra': 'ACACA'}, None)
+
+        #  reverse strand alignment for Insertion
+        # For insertion there cannot be a reference allele change on the negative strand because the reference allele
```

Comment:
Did you mean to write more in this sentence?

Reply:
Yes... I added it now . Thanks